File: Generation/JavaScript/Fundamentals/VersionInfo.js
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

var Fundamentals_Versioning_Version_pb = require('./Versioning/Version_pb');

function VersionInfo() {}

VersionInfo.prototype.getCurrentVersion = function() {
    return VersionInfo.getCurrentVersion();
}

VersionInfo.getCurrentVersion = function() {
    return new Fundamentals_Versioning_Version_pb.Version()
        .setMajor(100)
        .setMinor(200)
        .setPatch(300)
        .setPrereleasestring("LOCAL");
}

exports.VersionInfo = VersionInfo;

File: Versioning/VersionInfo.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Dolittle.Versioning.Contracts;

namespace Dolittle.Contracts
{
    /// <summary>
    /// Provides info about the current version of the Contracts.
    /// </summary>
    public static class VersionInfo
    {
        /// <summary>
        /// Gets the value indicating the current <see cref="Version"/> of the Contracts.
        /// </summary>
        public static Version CurrentVersion { get; } = new Version
        {
            Major = 5,
            Minor = 1,
            Patch = 0,
            PreReleaseString = "local",
        };
    }
}
